provider_config.go

```diff
@@ -0,0 +1,57 @@
+package jwtauth
+
+import (
+	"fmt"
+)
+
+// Provider-specific configuration interfaces
+// All providers must implement the CustomProvider interface, and may implement
+// others as needed.
+
+// ProviderMap is used to map a provider name to its provider type
+var ProviderMap = map[string]CustomProvider{
```

Comment:
nit/question: `ProviderMap` be a function, to discourage manipulating it?

Reply:
Discussed/resolved in https://github.com/hashicorp/vault-plugin-auth-jwt/pull/118#discussion_r440808605